screens/OrderItemDetailEditModal.js

```diff
@@ -0,0 +1,541 @@
+import RadioItemObjPick from '../components/RadioItemObjPick'
+import InputText from "../components/InputText";
+import RenderCheckBox from "../components/rn-elements/CheckBox";
+import Modal from 'react-native-modal';
+
+
+
+const OrderItemDetailEditModal = (props) => {
+    console.log("NAV?", props?.navigation)
+
+    const [modalVisible, setModalVisible] = useState(props?.modalVisible ?? false);
+
+    useEffect(() => {
+        setModalVisible(props?.modalVisible ?? false);
+        console.log("NewOrderModal", JSON.stringify(props?.data));
+    }, [props?.modalVisible]);
+
+    const handleSubmit = values => {
+        console.warn("handleSubmit");
+        const createOrder = {}
+        createOrder.orderType = 'IN_STORE'
+        createOrder.tableId = props?.data?.table?.tableId
+        createOrder.demographicData = {
+            male: values.male,
+            female: values.female,
+            kid: values.kid,
+            ageGroup: values.ageGroup,
+            visitFrequency: values.visitFrequency
+        }
+
+        dispatchFetchRequest(api.order.new, {
+            method: 'POST',
+            withCredentials: true,
+            credentials: 'include',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(createOrder)
+        },
+            response => {
+                response.json().then(data => {
+                    props?.submitOrder(data.orderId)
+                })
+            }).then()
+    }
+    return (
+
+        <Modal
+            isVisible={modalVisible}
+            backdropOpacity={0.7}
+            onBackdropPress={() => props.closeModal()}
+            useNativeDriver
+        >
+
+            <View style={{
+                borderRadius: 10,
+                width: '80%',
+                height: '90%',
+                borderWidth: 5,
+                borderColor: mainThemeColor,
+                marginTop: 53,
+                justifyContent: 'center',
+                marginBottom: 53,
+                alignSelf: 'center',
+            }}>
+                <KeyboardAvoidingView behavior='position' keyboardVerticalOffset={80}>
+                    <OrderFormIII
+                        onSubmit={handleSubmit}
+                        navigation={props.navigation}
+                        prdId={props.prdId}
+                        //tablesMap={tablesMap}
+                        initialValues={{
+                            male: 0,
+                            female: 0,
+                            kid: 0,
+                            data: props.data
+                        }}
+                        goBack={props.closeModal}
+                        isEditLineItem={props?.isEditLineItem ?? false}
+                        modalData={props?.data}
+                    />
+                </KeyboardAvoidingView>
+
+
+            </View>
+
+        </Modal>
+
+
+
+    );
+};
+
+class OrderFormIIIBase extends React.Component {
```

Comment:
Name suggestion: ConnectedOrderItemOptions

Reply:
named it ConnectedOrderItemOptionsBase